src/pages/en/guides/cms/ghost.mdx

```diff
@@ -2,12 +2,252 @@
+- src/
+  - lib/
+    - **api.ts**
+  - pages/
+    - **index.astro**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+```ts title="src/lib/api.ts"
+
+import GhostContentAPI from '@tryghost/content-api';
+
+// Create API instance with site credentials
+export const api = new GhostContentAPI({
+    url: 'http://localhost:2368', // This is the URL if your site is running on a local environment
+    key: import.meta.env.CONTENT_API_KEY,
+    version: 'v5.0',
+});
+
+```
+
+Here, we first initialize an API instance with the Ghost API. Using the API key we got from our dashboard's Integrations page.
+
+### Displaying a list of posts
+
+The page `src/pages/index.astro` lists each post, with a description and link to its own page.
+
+<FileTree title="Project Structure">
+- src/
+ - lib/
+    - api.ts
+  - pages/
+    - **index.astro**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+In the Astro frontmatter, we first import our API instance. This would allow us to use the `posts.browse()` method to brows the posts. We're setting the `limit` to `all` to get all of our blog posts.
+
+```astro title="src/pages/index.astro"
+---
+import { api } from '../lib/api';
+const posts = await api.posts
+    .browse({
+        limit: 'all',
+    })
+    .catch((err) => {
+        console.error(err);
+    });
+---
+```
+
+Fetching post via the content API returns an object that includes the properties:
+- `title` - Contains the title of the post
+- `html` - Contains the HTML rendering of the content of the post
+- `feature_image` - Contains the source URL of the featured image of the post
+- `slug` - Contains the slug of the post
+
+Now, we can render our posts to HTML.
+
+```astro title="src/pages/index.astro"
+<html lang="en">
+    <head>
+        <title>Astro + Ghost 👻</title>
+    </head>
+    <body>
+
+        {
+            posts.map((post) => (
+                <>
+                    <a href={`/post/${post.slug}`}>
+                        <h1> {post.title} </h1>
+                    </a>
+                </>
+            ))
+        }
+    </body>
+</html>
+```
+
+### Using the Ghost Content API to generate pages
+
+The page `src/pages/post/[slug].astro` [dynamically generates a page](/en/core-concepts/routing/#dynamic-routes) for each post.
+
+<FileTree title="Project Structure">
+- src/
+ - lib/
+    - api.ts
+  - pages/
+    - index.astro
+    - post/
+      - **[slug].astro**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+Just like the [index page](#displaying-a-list-of-posts), we'll first import our API instance. Following which, we'll have access to the `posts.browse()` method.
+
+Here, we're getting the posts and returning each single post in the [props](/en/tutorial/5-astro-api/2/#use-props-in-dynamic-routes).
+
+```astro title="src/pages/post/[slug].astro"
+---
+import { api } from '../../lib/api';
+
+// getStaticPaths() is required for static Astro sites
+// if using SSR, you will not need this function
```

Comment:
```suggestion
```
Let's split these up into 2 versions, like the other guides. If you're using SSG and if you're using SSR.